source/portable/NetworkInterface/xilinx_ultrascale/x_emacpsif_hw.c

```diff
@@ -47,17 +47,16 @@ void setup_isr( xemacpsif_s * xemacpsif )
     /*
      * Setup callbacks
      */
-    XEmacPs_SetHandler( &xemacpsif->emacps, XEMACPS_HANDLER_DMASEND,
-                        ( void * ) emacps_send_handler,
-                        ( void * ) xemacpsif );
+    XEmacPs * xInstancePtr = &xemacpsif->emacps;
 
-    XEmacPs_SetHandler( &xemacpsif->emacps, XEMACPS_HANDLER_DMARECV,
-                        ( void * ) emacps_recv_handler,
-                        ( void * ) xemacpsif );
+    xInstancePtr->SendHandler = ( XEmacPs_Handler ) emacps_send_handler;
+    xInstancePtr->SendRef = ( void * ) xemacpsif;
 
-    XEmacPs_SetHandler( &xemacpsif->emacps, XEMACPS_HANDLER_ERROR,
-                        ( void * ) emacps_error_handler,
-                        ( void * ) xemacpsif );
+    xInstancePtr->RecvHandler = ( XEmacPs_Handler ) emacps_recv_handler;
```

Comment:
Are these casts to XEmacPs_Handler and XEmacPs_ErrHandler necessary?

Reply:
For reference, https://github.com/Xilinx/embeddedsw/blob/master/XilinxProcessorIPLib/drivers/emacps/src/xemacps.h#L531 seems to list XEmacPs_Handler and XEmacPs_ErrHandler types for these fields.

Reply:
No, as far as I know the casts shouldn't be necessary. However I think it is a nice extra touch to do it them explicitly.

Reply:
I think it would be preferable to remove the explicit casts so that a compiler warning is generated if the definitions do not match up.

Reply:
+1 to Paul's comment above.
If anytime in future if we change the typedef, I would like the compiler to issue a warning (which cannot happen with a cast).
Would you mind making this change please?

Reply:
No at all, I am going to address the comment in a coming commit.